cmd/state-svc/service.go

```diff
@@ -82,3 +92,32 @@ func (s *service) Wait() error {
 	}
 	return nil
 }
+
+type resolveExec struct {
+	resolver *resolver.Resolver
+}
+
+func (r *resolveExec) ReportRuntimeUsage(ctx context.Context, pid, exec string) {
```

Comment:
I can see why it looks like corner cutting. The resolver has the method RuntimeUsage which is fairly generic in that it can accommodate both `state exec` and `state-exec` cases. The calling logic for `state exec` is in the relevant runner, but the case for `state-exec` doesn't have a nice home, so I placed it here to wait for your direction. In essence, this is a new scope (that correlates to a "runner" package) that isn't yet organized. Until there's more of a need, I am content with it being here.

Reply:
I don't see the current organization of the svc that way. Resolver is anything that handles requests. Whether those requests come in through the socket or through http ought to be irrelevant to the resolver package.

So I would still advocate we put it in the resolver package.